slider/beatmap.py

```diff
@@ -945,16 +946,15 @@ def _moving_average_by_time(times, data, delta, num):
 
     # append a nan to the end of the values so that we can do many slices all
     # the way to the end in reduceat
-    values = np.vstack([data, [np.nan, np.nan]])
+    values = np.vstack([data, [np.nan] * data.shape[1]])
 
     # sum the values in the ranges ``[window_start_ixs, window_stop_ixs)``
-    speed_window_sums = np.add.reduceat(values[:, 0], window_ixs.ravel())[::2]
-    aim_window_sums = np.add.reduceat(values[:, 1], window_ixs.ravel())[::2]
+    window_sums = (np.add.reduceat(i, window_ixs.ravel())[::2] for i in values.transpose())
```

Comment:
we can pass the 2d array directly to reduceat; we may need to also pass the axis we want to reduce across though it should default to 0

Reply:
yeah i pushed this kind of expecting you to explain some way of doing this better with numpy tbh but i didn't know enough about it to work it out myself :P